beater/journalbeat.go

```diff
@@ -146,6 +191,124 @@ func New(b *beat.Beat, cfg *common.Config) (beat.Beater, error) {
 	return jb, nil
 }
 
+//TODO optimize this later but for now walkthru all the different types.
+	func (jb *Journalbeat) logProcessor() {
+	tickChan := time.NewTicker(time.Second * 30).C
+	for {
+		select {
+			case <- tickChan:
+				//here we need to walk thru all the map entries and flush out the ones
+				//which have been sitting there for some time.
+				jb.flushStaleLogMessages()
+
+			case channelEvent := <- jb.incomingLogMessages:
+				jb.flushOrBufferLogs(channelEvent)
+		}
+	}
+}
+
+func (jb *Journalbeat) flushStaleLogMessages() {
+	for logType, logBuffer := range jb.journalTypeOutstandingLogBuffer {
```

Comment:
If there are a lot of logs this could be fairly inefficient. Can we push them to a channel or queue?